envoy/stats/sink.h

```diff
@@ -17,7 +17,7 @@ class Histogram;
 class MetricSnapshot {
 public:
   struct CounterSnapshot {
-    uint64_t delta_;
+    int64_t delta_;
```

Comment:
This is a breaking change? We cannot change the existing counter behavior.

Reply:
If the value no bigger than the max(int64_t), then this should not be a break change?

Reply:
+1. I really suspect no users really use counters beyond 2 ^ 63. 

If we can do this extension, it save lots of efforts to add new stats type(adding stats in core and also make stat_sink extensions support this stat)

Reply:
+1 to what Greg says. They should appear different to stat sinks since not all sinks can report delta updown counter and must convert them to gauges. We don't want to convert every counter to gauge.